src/main.c

```diff
@@ -1,43 +1,72 @@
-#include <libopencm3/stm32/rcc.h>
+#include <libopencm3/cm3/nvic.h>
+#include <libopencm3/cm3/systick.h>
 #include <libopencm3/stm32/gpio.h>
+#include <libopencm3/stm32/rcc.h>
 
 
-static void clock_setup(void)
+/**
+ * @brief Initial clock setup.
+ *
+ * Use the External High Speed clock (HSE), at 8 MHz, and set the SYSCLK
+ * at 72 MHz (the maximum allowed when using the external crystal/resonator).
+ * This output frequency is possible thanks to the Phase Locked Loop (PLL)
+ * multiplier.
+ *
+ * Enable required clocks for the GPIOs as well.
+ */
+static void setup_clock(void)
 {
-	/* Set STM32 to 72 MHz. */
-	rcc_clock_setup_in_hse_12mhz_out_72mhz();
-
-	/* Enable GPIOC, and AFIO clocks. */
+	rcc_clock_setup_in_hse_8mhz_out_72mhz();
```

Comment:
Why are you using now 8 mhz as input instead of the 12 mhz from previous commit?